src/chaosinventory/base/serializers.py

```diff
@@ -0,0 +1,143 @@
+from rest_framework import serializers
+
+from .models import (
+    DataType, Entity, Item, Location, LocationData, Product, ProductData, Tag,
+)
+
+
+class BasicLocationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Location
+        fields = [
+            'id',
+            'name',
+            'note',
+            'in_location'  # TODO: Make recursive?
+        ]
+
+
+class TagSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Tag
+        fields = [
+            'id',
+            'parent'
+        ]
+
+
+class BasicLocationDataSerializer(serializers.ModelSerializer):
+    type = serializers.ReadOnlyField(source='type.name')
+
+    class Meta:
+        model = LocationData
+        fields = [
+            'id',
+            'value',
+            'type'
+        ]
+
+
+class BasicProductDataSerializer(serializers.ModelSerializer):
+    type = serializers.ReadOnlyField(source='type.name')
+
+    class Meta:
+        model = ProductData
+        fields = [
+            'id',
+            'value',
+            'type'
+        ]
+
+
+class DataTypeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = DataType
+        fields = [
+            'id',
+            'name',
+            'note'
+        ]
+
+
+class LocationDataSerializer(serializers.ModelSerializer):
+    type = serializers.ReadOnlyField(source='type.name')
+    location = BasicLocationSerializer()
+
+    class Meta:
+        model = LocationData
+        fields = [
+            'id',
+            'value',
+            'type',
+            'location'
+        ]
+
+
+class LocationSerializer(serializers.ModelSerializer):
+    locationdata_set = BasicLocationDataSerializer(many=True, required=False)
+
+    class Meta:
+        model = Location
+        fields = [
+            'id',
+            'name',
+            'note',
+            'in_location',
+            'locationdata_set'
+        ]
+
+
+class EntitySerializer(serializers.ModelSerializer):
+    tags = TagSerializer(many=True, required=False)
+
+    class Meta:
+        model = Entity
+        fields = [
+            'name',
+            'note',
+            'part_of',  # TODO: Make recursive?
+            'tags',
+        ]
+
+
+class ProductSerializer(serializers.ModelSerializer):
+    tags = TagSerializer(many=True, required=False)
+
+    productdata_set = BasicLocationDataSerializer(
+        many=True,
+        required=False
+    )
+
+    class Meta:
+        model = Product
+        fields = [
+            'name',
+            'note',
+            'tags',
```

Comment:
In my test I was unable to create an Product with an already existing Tag. I tried to do this with the ID of the Tag. Maybe that's a bug or I need some explanation how to do this